.gitlab-ci.yml

```diff
@@ -6,42 +6,51 @@ default:
   image: ${CI_REGISTRY}/${CI_PROJECT_PATH}/lpf-ubuntu-20.04-gcc
 
 variables:
-  IS_MAIN_PIPELINE: "no"
+  EXTRA_TESTS_ENABLED: "no" # enable extra tests: build with debug symbols, performance tests
+  LPF_TESTS_ENABLED: "no" # enable LPF-based tests
+  MULTIPLE_COMPILERS_ENABLED: "no" # enable compiling against multiple compilers
 
 workflow:
   rules:
-  # the following rules trigger after new commits are merged on master, develop or a version/rc branch
+  # run all jobs if a commit is merged on master, develop or a version/rc branch
     - if: $CI_COMMIT_BRANCH == "master"
       variables:
-        IS_MAIN_PIPELINE: "yes"
+        EXTRA_TESTS_ENABLED: "yes"
+        LPF_TESTS_ENABLED: "yes"
+        MULTIPLE_COMPILERS_ENABLED: "yes"
     - if: $CI_COMMIT_BRANCH == "develop"
       variables:
-        IS_MAIN_PIPELINE: "yes"
+        EXTRA_TESTS_ENABLED: "yes"
+        LPF_TESTS_ENABLED: "yes"
+        MULTIPLE_COMPILERS_ENABLED: "yes"
     - if: $CI_COMMIT_BRANCH =~ /^v.*-rc.*$/
       variables:
-        IS_MAIN_PIPELINE: "yes"
-  # rules to avoid duplicate pipelines on merge requests
+        EXTRA_TESTS_ENABLED: "yes"
+        LPF_TESTS_ENABLED: "yes"
+        MULTIPLE_COMPILERS_ENABLED: "yes"
+    # "manual" execution strategies are allowed to force run
+    - if: $CI_PIPELINE_SOURCE != "merge_request_event" && $CI_PIPELINE_SOURCE != "push"
```

Comment:
Does a manual trigger of the pipeline require manually setting the `*_ENABLED` flags? (On Jenkins the GUI would expose the available flags and make it easy to change them, when manually triggering a run -- not sure if Gitlab allows the same?)

Reply:
Gitlab is able to recognise the variables and expose them in the interface. 
It is now possible, see [this commit](https://github.com/Algebraic-Programming/ALP/pull/200/commits/530e2a1e4d3a373c97f41d2ae05dc88dc72616e9)